app/controllers/attributes_controller.rb

```diff
@@ -0,0 +1,38 @@
+class AttributesController < ApplicationController
+  load_and_authorize_resource except: [:index]
+  before_action :set_type
+  respond_to :js, :html
+
+  def index
+    @attributes = @type.character_attributes.order('created_at DESC').page(params[:page]).per(4)
+  end
+
+  def create
+    @attribute = @type.character_attributes.new(attribute_params)
+    saved = @attribute.save
+    render :new unless saved
+  end
+
+  def update
+    updated = @attribute.update(attribute_params_update)
+    render :edit unless updated
+  end
+
+  def destroy
+    @attribute.destroy
+  end
+
+  private
+
+  def set_type
+    @type = Type.find(params[:type_id])
+  end
+
+  def attribute_params
+    params.require(:attribute).permit(:name, :numerical, :type_id, :icon)
+  end
+
+  def attribute_params_update
```

Comment:
@VanjaJeftic if required/allowed params are the same there’s no need to have two separate methods.

Reply:
@basti  I try to find an earlier conversation where you said to me that I need a separate params method in type controller, but I can't find it. So I thought that I need to separate params in the application controller too?

Reply:
@VanjaJeftic if I recall correctly I told you that because you had a conditional that depended on called action. Now they are the same, so there’s no need to have separate param methods.